test/architecture/wait-for-architecture.sh

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+
+THIS_DIR=$(cd -P -- "$(dirname -- "$0")" && pwd -P)
+TEST_DIR="${THIS_DIR}/.."
+REPO_ROOT_DIR="${TEST_DIR}/.."
+DEFAULT_SG_CONFIG_FILE="${TEST_DIR}/resources/.sgconfig"
+
+export SG_CONFIG_FILE=${SG_CONFIG_FILE-"${DEFAULT_SG_CONFIG_FILE}"}
+
+echo "Wait for architecture to be up"
+echo "Using config file at $SG_CONFIG_FILE ..."
+
+_init_engines() {
+  ( SG_ENGINE_HOST=local_engine SG_ENGINE_PORT=5432 sgr init ) \
+    && ( SG_ENGINE_HOST=remote_engine SG_ENGINE_PORT=5431 sgr init ) \
+    && return 0
+
+  return 1
+}
+
+_run_health_check() {
+    pushd "$REPO_ROOT_DIR" \
+        && python -c "import test.splitgraph.conftest as c; c.healthcheck()" \
```

Comment:
I wonder if it's worth spitting the CI jobs to run the MySQL integration test (the single one) separately since IIRC waiting for it to start up is what delays `wait_for_architecture()`. In fact, most tests currently rely on data from fixtures etc rather than from `pg`/`mongo`/`mysqlorigin` so we could even move those to the end of the suite and start the test dbs in parallel with running the rest of the tests.